### PR TITLE
Culture specific bugs are fixed. All tests are "green" now.

### PR DESCRIPTION
**Pull request by psmolich79**
Fixed problems related to culture and proper conversions.

### CODE REVIEW

**Comment by iwate** [comment]
Hi @psmolich79  ! Thank you for your contribute!

I check your code and it looks good :)

However, I want to know what values cause error.

Would you add a few test cases specific values ?

**Comment by bxbxbx** [comment]
Hi @iwate please consider to merge this pull request, as it fixes formatting problems on non english systems.
The source of the problems is that the actual culture is used in the parameterizers, which leads to problems with floating point numbers on non english systems.

File: src/ODataHttpClient/Models/Request.cs [hunk 1]
```diff
@@ -57,7 +57,8 @@ public static Request Create<T>(HttpMethod method, string uri, T body, IEnumerab
 
             if (type.IsValueType || type == stringType)
             {
-                content = body.ToString();
+                //content = body.ToString();
```
Comment (iwate) [inline src/ODataHttpClient/Models/Request.cs:60]: This need plain text, so we cannot use json serializer.

**Comment by iwate** [comment]
@bxbxbx Thank you for your comments.

I understand this current culture issue but I cannot merge this PR because this patch has a many changes from current master.

I've added red pattern  test codes for this issue. If @psmolich79 modify and recreate this PR, I merge it.

When this PR is not update until next weekend, I create a patch by self and close this PR.

https://github.com/iwate/ODataHttpClient/commit/597337a80d4b00be997eca4198f390be2e9b9e9e

**Comment by psmolich79** [comment]
> @bxbxbx Thank you for your comments.
> 
> I understand this current culture issue but I cannot merge this PR because this patch has a many changes from current master.
> 
> I've added red pattern test codes for this issue. ([597337a](https://github.com/iwate/ODataHttpClient/commit/597337a80d4b00be997eca4198f390be2e9b9e9e)) If @psmolich79 could modify and recreate this PR, I merge it.
> 
> When this PR is not update until next weekend, I create a patch by self and close this PR.

Hi. There are too many changes due to my habit of cleaning the code with Resharper :) Sorry about that. At the beginning, I didn't plan to ask for merge of my changes. I'll try to sort it out and make only a minimal amount of changes this weekend.

**Comment by psmolich79** [comment]
Hi @iwate . Please check if this pull request can be integrated now.

File: src/ODataHttpClient/Models/Request.cs [hunk 2]
```diff
@@ -73,9 +73,14 @@ public static Request Create<T>(HttpMethod method, string uri, T body, IEnumerab
...
-            if (type.IsValueType || type == stringType)
+            if (type == stringType)
             {
-                content = body.ToString();
+                content = body.ToString(); 
+                mime = "text/plain";
+            }
+            else if (type.IsValueType)
+            {
+                content = serializer.Serialize<T>(body);
```
Comment (iwate) [inline src/ODataHttpClient/Models/Request.cs:83]: This request content type is needed plain text whichi cannnot use serializer. use `ToString` with CultureInfo.InvariantCulture.

File: src/ODataHttpClient/Models/Response.cs [hunk 3]
```diff
@@ -16,10 +14,8 @@ public class Response
         public string MediaType { get; private set; }
         public string ErrorMessage { get; private set; }
         public byte[] Binary { get; private set; }
-        public string Body { get => GetStringFromUTF8(Binary); }
```
Comment (iwate) [inline src/ODataHttpClient/Models/Response.cs:19]: GetStringFromUTF8 is needed for BOM

**Review by iwate — CHANGES_REQUESTED** [review]
@psmolich79 Thank you modify PR. 

However, this change includes a change that reverts to the old code.

Could you re-create it including only your changes?

If it is difficult, you can create new a branch from current master and create new a PR.

**Comment by iwate** [comment]
This issue is resolved at #24

**Pull request closed without merging**